src/cpp/include/openvino/genai/rag/text_embedding_pipeline.hpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <filesystem>
+#include <optional>
+#include <variant>
+
+#include "openvino/genai/tokenizer.hpp"
+
+namespace ov {
+namespace genai {
+
+using EmbeddingResult = std::variant<std::vector<float>, std::vector<int8_t>, std::vector<uint8_t>>;
+
+class OPENVINO_GENAI_EXPORTS TextEmbeddingPipeline {
+public:
+    enum class OPENVINO_GENAI_EXPORTS PoolingType {
```

Comment:
```suggestion
    enum class PoolingType {
```
`export` is not required for classes / enums, etc which don't have any methods in .cpp files

Reply:
export removed